packages/labs/src/components/omnibox/omnibox.tsx

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+
+import {
+    AbstractComponent,
+    Classes as CoreClasses,
+    HTMLInputProps,
+    IBackdropProps,
+    IInputGroupProps,
+    InputGroup,
+    IOverlayableProps,
+    IOverlayProps,
+    Overlay,
+    Utils,
+} from "@blueprintjs/core";
+import {
+    IListItemsProps,
+    IQueryListRendererProps,
+    ISelectItemRendererProps,
+    QueryList,
+} from "../";
+import * as Classes from "../../common/classes";
+
+export interface IOmniboxProps<T> extends IListItemsProps<T> {
+    /**
+     * Custom renderer for an item in the dropdown list. Receives a boolean indicating whether
+     * this item is active (selected by keyboard arrows) and an `onClick` event handler that
+     * should be attached to the returned element.
+     */
+    itemRenderer: (itemProps: ISelectItemRendererProps<T>) => JSX.Element;
+
+    /** React child to render when filtering items returns zero results. */
+    noResults?: string | JSX.Element;
+
+    /**
+     * Props to spread to `InputGroup`. All props are supported except `ref` (use `inputRef` instead).
+     * If you want to control the filter input, you can pass `value` and `onChange` here
+     * to override `Select`'s own behavior.
+     */
+    inputProps?: IInputGroupProps & HTMLInputProps;
+
+    /**
+     * Toggles the visibility of the omnibox.
+     * This prop is required because the component is controlled.
+     */
+    isOpen: boolean;
+
+    /**
+     * A callback that is invoked when user interaction causes the overlay to close, such as
+     * clicking on the overlay or pressing the `esc` key (if enabled).
+     * Receives the event from the user's interaction, if there was an event (generally either a
+     * mouse or key event). Note that, since this component is controlled by the `isOpen` prop, it
+     * will not actually close itself until that prop becomes `false`.
+     */
+    onClose?: (event?: React.SyntheticEvent<HTMLElement>) => void;
+
+    /** Props to spread to `Overlay`. Note that `content` cannot be changed. */
+    overlayProps?: Partial<IOverlayProps>;
```

Comment:
Make the type `Partial<IOverlayProps> & object` for strict type-checking. (Otherwise, you can pass in stuff like `{ overlayProps: "not an object" }`, and it won't complain!